engine/src/resource/resource.cpp

```diff
@@ -31,6 +31,49 @@
 
 #include "random_utils.h"
 
+
+template<typename T>
+T _convert(const std::string input, const T modifier);
+
+template<>
+float _convert<float>(const std::string input, const float modifier) {
+    try {
+        return std::stod(input) * modifier;
+    } catch(...) {
+        return 0.0;
+    }
+}
```

Comment:
this should return `float` types not `double` types. So `0.0f` instead of `0.0` and `std::stof()` instead of `std::stod()`

Reply:
Good catch. I wonder if the compiler is smart enough to optimize this.

Reply:
it might catch L43, but it can't do the conversion of L41 automatically; though it might treat it like L43. Still, better to just to the right thin to start with. Thanks for fixing!